src/compile/unit.ts

```diff
@@ -234,10 +234,17 @@ export class UnitModel extends Model {
 
         // We no longer support false in the schema, but we keep false here for backward compatability.
         if (axisSpec !== null && axisSpec !== false) {
-          _axis[channel] = extend({},
-            config.axis,
-            axisSpec === true ? {} : axisSpec ||  {}
-          );
+          let vlAxisBase: VlAxisBase = {};
+          AXIS_BASE_PROPERTIES.forEach(function(property) {
+            if (encoding !== undefined && encoding[property] !== undefined) {
+              vlAxisBase[property] = encoding[property];
+            }
+          });
+          _axis[channel] = {
+            labelMaxLength: config.axis.labelMaxLength,
+            vlAxisBase,
```

Comment:
1)  It should be `...vlAxisBase`

2) You didn't write unit test for this part.